tensorflow/c/experimental/filesystem/plugins/gcs/gcs_filesystem_test.cc

```diff
@@ -12,50 +12,150 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
+#include <random>
+
+#include "google/cloud/storage/client.h"
 #include "tensorflow/c/experimental/filesystem/filesystem_interface.h"
 #include "tensorflow/c/tf_status_helper.h"
+#include "tensorflow/core/platform/path.h"
 #include "tensorflow/core/platform/stacktrace_handler.h"
 #include "tensorflow/core/platform/test.h"
 
-#define ASSERT_TF_OK(x) ASSERT_EQ(TF_OK, TF_GetCode(x))
+#define ASSERT_TF_OK(x) ASSERT_EQ(TF_OK, TF_GetCode(x)) << TF_Message(x)
+
+static const char* content = "abcdefghijklmnopqrstuvwxyz1234567890";  // 36
+
+namespace gcs = google::cloud::storage;
 
 // Forward declaration
+void ParseGCSPath(absl::string_view fname, bool object_empty_ok, char** bucket,
+                  char** object, TF_Status* status);
+
+namespace tf_random_access_file {
+void Cleanup(TF_RandomAccessFile* file);
+int64_t Read(const TF_RandomAccessFile* file, uint64_t offset, size_t n,
+             char* buffer, TF_Status* status);
+}  // namespace tf_random_access_file
```

Comment:
At this point it seems you really need a header file for the stuff that gets exposed to testing, since you are doing so many forward declarations duplicating the code.

It's ok if more symbols get exposed from the plugin, TF won't call them directly. It's much better than the decrease in readability and the potential of having the forward declarations here not match the ones in the plugin itself.

Let's make a PR which creates the header file and cleans up these forward declarations and the `STATIC` macro.